### ISSUE TITLE
Applications Folder

### ISSUE THREAD

**Issue by smrf1093** [cookiecutter/cookiecutter-django#3594]
## Description

Move all the application folders into a subfolder named apps 

## Rationale

It is more logical to have all Django applications in one subfolder, here is a StackOverflow discussion which suggests this
https://stackoverflow.com/questions/22841764/best-practice-for-django-project-working-directory-structure

**Comment by Andrew-Chen-Wang** [comment]
Isn't it already under one sub folder called the "project_slug"?

**Comment by browniebroke** [comment]
@smrf1093 would you find it less confusing if folders for non-apps were outside of the `project_slug` folder? I kind of see what you mean, but at the same time I don't really see the need for another level called `apps`. The main drawback I see is that it would increase the length of all the imports:

```diff
- from project_slug.users.forms import ...
+ from project_slug.apps.users.forms import ...
```

Although on a second thought, I would like to leave the `utils` package where it is, as it enables us to easily namespace isort's known 1st party:

https://github.com/cookiecutter/cookiecutter-django/blob/d2f129e3447e4870f9bb84096981db7f89c8cd8a/%7B%7Bcookiecutter.project_slug%7D%7D/setup.cfg#L11

**Comment by smrf1093** [comment]
@browniebroke  I nearly agree with your suggestion at least we know all the apps are placed inside the project_slug folder and this scope is restricted to the Django project applications. about the lengthy imports, I do not completely agree with you because just adding 4 characters to the imports does not make them lengthy besides it adds meaning to it for example when you write project_slug.apps.devices it means that the devices package contains a Django application

**Comment by browniebroke** [comment]
Granted, it's not many characters on a single line, but it would be repeated many times, for nearly all internal imports of the project. I think that cost is not worth it as I see no benefit (at least so far).

**Comment by huydbui** [comment]
> @browniebroke I nearly agree with your suggestion at least we know all the apps are placed inside the project_slug folder and this scope is restricted to the Django project applications. about the lengthy imports, I do not completely agree with you because just adding 4 characters to the imports does not make them lengthy besides it adds meaning to it for example when you write project_slug.apps.devices it means that the devices package contains a Django application

These implementations are more like user preferences, you don't need to wait for them. I have been using it for years. All of my apps (just my apps) are in an "apps" (or similar name) folder.

**Comment by smrf1093** [comment]
> > @browniebroke I nearly agree with your suggestion at least we know all the apps are placed inside the project_slug folder and this scope is restricted to the Django project applications. about the lengthy imports, I do not completely agree with you because just adding 4 characters to the imports does not make them lengthy besides it adds meaning to it for example when you write project_slug.apps.devices it means that the devices package contains a Django application
> 
> These implementations are more like user preferences, you don't need to wait for them. I have been using it for years. All of my apps (just my apps) are in an "apps" (or similar name) folder. I mostly use relative imports between apps, so I don't see the "apps." in my imports at all.

Yes, I know it is a user preference. However, many of the cookie-cutter structures can be changed based on user preferences, but the purpose of it is to present a good starting structure. So, based on the purpose it is better to have this feature included.

**Comment by foarsitter** [comment]
Related to https://github.com/cookiecutter/cookiecutter-django/issues/4031
